core/src/main/scala/sttp/tapir/server/internal/DecodeInputs.scala

```diff
@@ -32,7 +32,7 @@ object DecodeInputsResult {
 
     def setBasicInputValue(v: Any, i: Int): Values = copy(basicInputsValues = basicInputsValues.updated(i, v))
   }
-  case class Failure(input: EndpointInput.Basic[_], failure: DecodeResult.Failure) extends DecodeInputsResult
+  case class Failure(input: EndpointInput.Basic[_], failure: DecodeResult.Failure, rootInput: EndpointInput[_]) extends DecodeInputsResult
```

Comment:
If I understand correctly and `rootInput` is always `e.input`, maybe we could simplify, and remove the `rootInput` from here & `InputValues`, adding this information *after* decoding, only in case of failure? It seems that threading this through all of these methods isn't necessary.

Reply:
it should be `e.input` always indeed. I've cleaned it up a little bit, thanks.